integration-tests/actions/ocr2_helpers_local.go

```diff
@@ -127,8 +129,38 @@ func CreateOCRv2JobsLocal(
-				ocrSpec.OCR2OracleSpec.RelayConfig["codec"] = `{"chainCodecConfig" :{"MedianReport":{"TypeAbi": "[{\"Name\": \"Timestamp\",\"Type\": \"uint32\"},{\"Name\": \"Observers\",\"Type\": \"bytes32\"},{\"Name\": \"Observations\",\"Type\": \"int192[]\"},{\"Name\": \"JuelsPerFeeCoin\",\"Type\": \"int192\"}]"}}}`
+				ocrSpec.OCR2OracleSpec.RelayConfig["chainReader"] = evmtypes.ChainReaderConfig{
+					ChainContractReaders: map[string]evmtypes.ChainContractReader{
```

Comment:
this ocr spec ends up marshalling chainReader and codec as invalid toml

`chainReader = {map[median:{[{"anonymous":false,"inputs":[{"indexed":true,"internalType":"address","name":"requester","type":"address"},{"indexed":false,"internalType":"bytes32","name":"configDigest","type":"bytes32"},{"indexed":false,"internalType":"uint32","name":"epoch","type":"uint32"},{"indexed":false,"internalType":"uint8","name":"round","type":"uint8"}],"name":"RoundRequested","type":"event"},{"inputs":[],"name":"latestTransmissionDetails","outputs":[{"internalType":"bytes32","name":"configDigest","type":"bytes32"},{"internalType":"uint32","name":"epoch","type":"uint32"},{"internalType":"uint8","name":"round","type":"uint8"},{"internalType":"int192","name":"latestAnswer_","type":"int192"},{"internalType":"uint64","name":"latestTimestamp_","type":"uint64"}],"stateMutability":"view","type":"function"}] map[LatestRoundRequested:{false RoundRequested event [] []} LatestTransmissionDetails:{false latestTransmissionDetails method [] [0x140013428c8]}]}]}
codec = {map[MedianReport:{[{"Name": "Timestamp","Type": "uint32"},{"Name": "Observers","Type": "bytes32"},{"Name": "Observations","Type": "int192[]"},{"Name": "JuelsPerFeeCoin","Type": "int192"}] []}]}
`

Reply:
in tests it is done with toml marshall, while in integration tests it is marshalled to string with marshallTemplate

Reply:
What is that? Can you link?

Reply:
[MarshallTemplate] (https://github.com/smartcontractkit/chainlink/blob/529d2cf8af1de371ba900b5f371496942ed52dc5/integration-tests/client/chainlink_models.go#L1305) used in [(o *OCR2TaskJobSpec) String() ](https://github.com/smartcontractkit/chainlink/blob/529d2cf8af1de371ba900b5f371496942ed52dc5/integration-tests/client/chainlink_models.go#L1025) which is used in integration tests in [CreateJob](https://github.com/smartcontractkit/chainlink/blob/529d2cf8af1de371ba900b5f371496942ed52dc5/integration-tests/client/chainlink.go#L123)

Reply:
Oh yeah I think that is just making a bad assumption that there will only ever be top level fields, and never any structured data. I'll take a close look.